src/symmetry.jl

```diff
@@ -200,13 +207,31 @@ function accumulate_over_symops!(ρaccu, ρin, basis, symops, Gs)
     ρaccu
 end
 
+# Low-pass filters ρ (in Fourier) so that symmetry operations acting on it stay in the grid
+function lowpass_for_symmetry!(ρ, basis, symops=basis.symops)
+    for (S, τ) in symops
+        for (ig, G) in enumerate(G_vectors(basis))
+            if index_G_vectors(basis, S * G) === nothing
+                ρ[ig] = 0
+            end
+        end
+    end
+    ρ
+end
+
 """
-Symmetrize a `RealFourierArray` by applying all symmetry operations of
-the basis (or all symmetries passed as the second argument) and forming
-the average.
+Symmetrize a `RealFourierArray` by applying all the model symmetries (by default) and forming the average.
 """
-function symmetrize(ρin::RealFourierArray, symops=symmetry_operations(ρin.basis))
-    ρout_fourier = accumulate_over_symops!(zero(ρin.fourier), ρin.fourier, ρin.basis, symops,
+function symmetrize(ρin::RealFourierArray{Tr, T}, symops=ρin.basis.model.symops) where {Tr, T}
+    ρin_fourier = copy(ρin.fourier)
+    lowpass_for_symmetry!(ρin_fourier, ρin.basis, symops)
+    ρout_fourier = accumulate_over_symops!(zero(ρin_fourier), ρin_fourier, ρin.basis, symops,
                                            G_vectors(ρin.basis)) ./ length(symops)
-    from_fourier(ρin.basis, ρout_fourier)
+    from_fourier(ρin.basis, ρout_fourier; assume_real=(T <: Real))
+end
+
+function check_symmetric(ρin::RealFourierArray, tol=1e-10, symops=ρin.basis.model.symops)
```

Comment:
I'd make these kwargs.

Reply:
Yes. Basically let's just replace all default arguments with kwargs, no?

Reply:
Basically yes. I'm sure there are exceptions, also in our code. But in general I would agree to that statement.